Archive/Draft/RFCxxxx-PS7-Bootstrapper.md

```diff
@@ -0,0 +1,256 @@
+    - This means that if PS7 is already in the `PATH`, then that version (whether installed via MSI, MSIX, or other means)
+      needs to be updated outside of the bootstrapper
+  - Otherwise, [install](#installer) PS7 and then pass-through all arguments to the new detached `pwsh` child process
+
+The boostrapper should explicitly create a detached child `pwsh` process and then exit.
+As a detached child process, it will continue execution even after the parent has exited.
+
+Expectation is that the bootstrapper will always be first in the `$env:PATH` and executed first, so it can also serve
+to [auto-update](#auto-updating) PS7.
+
+This will allow `pwsh -c customer_script.ps1` to work seamlessly as though PS7 was available inbox.
+
+The bootstrapper would need to start the real `pwsh.exe` and bind stdin/stdout so that inputs and outputs come from `pwsh.exe` seamlessly.
+Unlike Unix, there isn't a way to `exec` a new process that replaces itself so for the first execution, there would be two `pwsh.exe` processes.
+
+However, if the bootstrapper detects that stdin, stdout, or stderr is redirected, then it would need to not exit and instead
+continue to redirect until the child `pwsh.exe` process has exited.
+
+Additional Windows requirements:
+
+- registered with Windows so that `Start->Run->pwsh` will start the bootstrapper
+- a separate Windows feature that can be uninstalled (so that using `pwsh` would not automatically work, or is being installed by other means)
+- different icon for the bootstrapper to indicate it is a bootstrapper
+
+### .ps1 registration
+
+`pwsh.exe` will be registered to execute .ps1 scripts, by default.
+However, it will have code to detect if started from the `explorer.exe` and instead
+give an error message that scripts need to be run from the console and not from Explorer.
+This means that from cmd.exe, you can execute .ps1 scripts without explicitly calling `pwsh.exe` and that applications
+can start depending on .ps1 scripts instead of wrapping them in batch files.
+The default execution policy on both Client and Server SKUs should be RemoteSigned.
+
+### Native code
+
+We can't rely on .NET Framework being available on some Windows skus we want to support and keeping the bootstrapper as
+small as possible, we don't want to rely on .NET (Core) and ship that runtime.
+Also, we want the bootstrapper to have minimal dependencies as future Windows SKUs may not ship with certain dependencies.
+
+#### Rust
+
+Rust has the benefit over C/C++ of having well defined contracts for resource management thus not having issues that persist in C/C++ code
+with regards to resource management which results in not only leaks but security issues.
+Resulting Rust binary code is very similar in size and performance as C/C++.
+Rust built code is supported in Windows.
+
+Rust uses the MS C runtime at execution, but we should statically link so that there aren't external dependencies that may be missing on the system.
```

Comment:
Statically linking the CRT will prevent servicing should the need arise (e.g. security). Instead, I would look into the _hybrid CRT_ solution. https://github.com/microsoft/WindowsAppSDK/blob/main/docs/Coding-Guidelines/HybridCRT.md